flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/util/RetryableExceptionClassifier.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.util;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.ExceptionUtils;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/** Classifier class for retryable exceptions on request submission failure. */
+@Internal
+public class RetryableExceptionClassifier {
+    private final Function<Throwable, Exception> throwableMapper;
+    private final Predicate<Throwable> validator;
+    private RetryableExceptionClassifier chainedClassifier;
+
+    public RetryableExceptionClassifier(
+            Predicate<Throwable> validator, Function<Throwable, Exception> throwableMapper) {
+        this.throwableMapper = throwableMapper;
+        this.validator = validator;
+        this.chainedClassifier = null;
+    }
+
+    public boolean shouldSuppress(Throwable err, Consumer<Exception> throwableConsumer) {
```

Comment:
Semantically this method is confusing. Since this is a `RetryableExceptionClassifier` would it be better to invert this to `shouldRetry`. Thoughts?

Reply:
I agree, we have been trying to align the name with a general function of exception classification and its current use case of validating retryable exceptions.
maybe `FatalExceptionClassifier` and `isFatal` would serve the purpose better.